rholang/src/main/scala/coop/rchain/rholang/interpreter/Interpreter.scala

```diff
@@ -0,0 +1,103 @@
+package coop.rchain.rholang.interpreter
```

Comment:
I have to say that I don't understand at all the point of this refactor. What are you trying to accomplish that wasn't solved by the previous implementation. This looks like moving things around just to look busy.

Reply:
It's about splitting the responsibilities and keeping as few responsibilities per class as possible, the ideal being of having one responsibility per class.

**Before the refactoring:**
RholangCLI had the following responsibilities:
1. read command line parameters
2. take action according to the command line parameters i.e. choose whether to  eval or repl or to write to binary or text etc.
3. the REPL logic (display prompt, read command, call evaluation, display results, repeat etc)
4. the eval logic (call evaluation, wait for task to finish, display results)
5. calling the whole execution flow (i.e. lexer+parser+normalizer+evaluation)

When somebody wanted to execute some Rholang code they had to execute RholangCLI.buildNormalizedTerm & friends. This is not intuitive since people would assume that the CLI stands for "command line interface"

**After this refactoring:**
* RholangCLI has responsibilities for the above items 1,2,3,4
* The responsibility no. 5 has been moved to Interpreter. 
* The Interpreter doesn't do any printing of the results. The caller does the printing so the interpreter can be used in an environment where we don't want printing.
* the code duplication was reduced in RholangCLI

Reply:
But this doesn't at all match the title or Summary of your pull request, so in your PR, you've violated the principle that you're trying to achieve here.

Reply:
So I first moved around the Rholang files as the Jira issue required.

Then, I had to fix the failing tests by adding the logic for running the rholang code in the interpreter. That was mostly a copy paste from some places where those functions were used. 

Third, I did a refactor of the things that I touched and left them in a better shape than they originally were.

Then I also had to do a few minor fixes as there were some tests failing. 

I guess next time I'll have to be more explicit in the PR description.